contracts/estate/EstateRegistry.sol

```diff
@@ -0,0 +1,236 @@
+  function safeTransferFrom(address, address, uint256) public;
+}
+
+
+// @nico TODO: Standalone Storage.
+// @nico TODO: Document functions.
+contract EstateRegistry is ERC721Token, Ownable, MetadataHolderBase {
+  using SafeMath for uint256;
+
+  PingableDAR public dar;
+
+  // From Estate to list of owned tokenIds (LANDs)
+  mapping(uint256 => uint256[]) public estateTokenIds;
+
+  // From tokenId (LAND) to its owner estateId of owned tokenIds
+  mapping(uint256 => uint256) public tokenIdEstate;
+
+  // From Estate to mapping of tokenId to index on the array above
+  mapping(uint256 => mapping(uint256 => uint256)) public estateTokenIndex;
+
+  // Metadata of the Estate
+  mapping(uint256 => string) internal estateData;
+
+  // Operators of the Estate
+  mapping (uint256 => address) public updateOperator;
+
+  constructor(
+    string _name,
+    string _symbol,
+    address _dar
+  )
+    ERC721Token(_name, _symbol)
+    Ownable()
+    public
+  {
+    require(_dar != 0);
+    dar = PingableDAR(_dar);
+  }
+
+  modifier onlyDAR() {
+    require(msg.sender == address(dar));
+    _;
+  }
+
+  modifier onlyAuthorized(uint256 estateId) {
+    require(_isAuthorized(msg.sender, estateId), "Unauthorized user");
+    _;
+  }
+
+  modifier onlyUpdateAuthorized(uint256 estateId) {
+    require(_isUpdateAuthorized(msg.sender, estateId), "Unauthorized user");
+    _;
+  }
+
+  function mintNext(address to) public returns (uint256) {
+    uint256 tokenId = _getNewTokenId();
+    _mint(to, tokenId);
+    return tokenId;
+  }
+
+  // onERC721Received: Count
+  function onERC721Received(
+    address /* oldOwner */,
+    uint256 tokenId,
+    bytes estateTokenIdBytes
+  )
+    external
+    onlyDAR
+    returns (bytes4)
+  {
+    if(estateTokenIdBytes.length > 0) {
+      uint256 estateId = uint256(_bytesToBytes32(estateTokenIdBytes));
+      _pushTokenId(estateId, tokenId);
+    }
+    return bytes4(0xf0b9e5ba);
+  }
+
+  function ammendReceived(uint256 estateId, uint256 tokenId) external {
+    _pushTokenId(estateId, tokenId);
+  }
+
+  function transferToken(
+    uint256 estateId,
+    uint256 tokenId,
+    address destinatary
+  )
+    onlyAuthorized(estateId)
+    external
+  {
+    return _transferToken(estateId, tokenId, destinatary);
+  }
+
+  function transferManyTokens(uint256 estateId, uint256[] tokens, address destinatary) onlyAuthorized(estateId) external {
+    uint length = tokens.length;
+    for (uint i = 0; i < length; i++) {
+      _transferToken(estateId, tokens[i], destinatary);
+    }
+  }
+
+  function getTokenEstateId(uint256 tokenId) external view returns(uint256) {
+    return tokenIdEstate[tokenId];
+  }
+
+  function ping() onlyOwner external {
+    dar.ping();
+  }
+
+  function getSize(uint256 estateId) external view returns (uint256) {
+    return estateTokenIds[estateId].length;
+  }
+
+  function updateMetadata(uint256 estateId, string metadata) external onlyUpdateAuthorized(estateId) {
+    estateData[estateId] = metadata;
```

Comment:
Missing event?

Reply:
Maybe we can start using the [ERC721Metadata](https://openzeppelin.org/api/docs/token_ERC721_ERC721_ERC721Metadata.html) standard for estates

Reply:
@nachomazzara 
Thing is that LANDRegistry forwards the updateMetadata method to the EstateRegistry contract. So I don't know if we can change it without breaking something in the process.
cc: @abarmat 

Adding `event` in the meantime

Reply:
I made a proposal which avoids calling EstateRegistry 's `updateMetada` from `LANDRegistry` contract